platform/openide.awt/src/org/openide/awt/DropDownButton.java

```diff
@@ -152,22 +159,33 @@ public void mouseExited( MouseEvent e ) {
     private PopupMenuListener getMenuListener() {
         if( null == menuListener ) {
             menuListener = new PopupMenuListener() {
+                @Override
                 public void popupMenuWillBecomeVisible(PopupMenuEvent e) {
                 }
 
+                @Override
                 public void popupMenuWillBecomeInvisible(PopupMenuEvent e) {
-                    // If inside the button let the button's mouse listener
-                    // deal with the state. The popup menu will be hidden and
-                    // we should not show it again.
                     if( getModel() instanceof Model ) {
                         ((Model)getModel())._release();
                     }
                     JPopupMenu menu = getPopupMenu();
                     if( null != menu ) {
                         menu.removePopupMenuListener( this );
                     }
+                    /* If the popup was closed by a mouse click inside the button area, the button
+                    may also receive a mousePressed event, although this seems not to be guaranteed.
+                    Ignore any such button press while the popup is closing, to avoid interpreting
+                    the press as a click to open the menu again. */
+                    popupClosingInProgress = true;
+                    SwingUtilities.invokeLater(new Runnable() {
```

Comment:
Lambda?

Reply:
Ah, sure. Didn't realize they existed in Java 8.